downloader/src/main/kotlin/VcsHost.kt

```diff
@@ -44,6 +46,52 @@ enum class VcsHost(
      */
     vararg supportedTypes: VcsType
 ) {
+    AZURE_DEVOPS("dev.azure.com", VcsType.GIT) {
+        private val commitPrefix = "GC"
+
+        override fun getUserOrOrgInternal(projectUrl: URI) = projectUrlToUserOrOrgAndProject(projectUrl)?.first
+
+        override fun getProjectInternal(projectUrl: URI) = projectUrl.path.substringAfterLast("/")
+
+        override fun toVcsInfoInternal(projectUrl: URI): VcsInfo {
+            val url = projectUrl.toString().replace(projectUrl.query.toString(), "").dropLast(1)
```

Comment:
So, this simply removes any query string, right? If so, I'd prefer to make the more clear by simply reconstructing an URI like

    URI(it.scheme, it.userInfo, it.host, it.port, it.path, null, it.fragment)